`-b` option is not working well after #819

```
LoadError: no such file to load -- bundler/dep_proxy
  block in expand_dependencies at uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:619
                          each at org/jruby/RubyArray.java:1734
  block in expand_dependencies at uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:618
                          each at org/jruby/RubyArray.java:1734
           expand_dependencies at uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:615
         converge_locked_specs at uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:583
                       resolve at uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:195
                         specs at uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:140
                     specs_for at uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:185
               requested_specs at uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:174
               requested_specs at uri:classloader:/gems/bundler-1.10.6/lib/bundler/environment.rb:18
                         setup at uri:classloader:/gems/bundler-1.10.6/lib/bundler/runtime.rb:13
                         setup at uri:classloader:/gems/bundler-1.10.6/lib/bundler.rb:127
                        <main> at uri:classloader:/gems/bundler-1.10.6/lib/bundler/setup.rb:8
                       require at org/jruby/RubyKernel.java:955
                       require at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:59
                        <main> at <script>:1
Exception in thread "main" com.google.inject.CreationException: Unable to create injector, see the following errors:

1) Error in custom provider, org.jruby.embed.EvalFailedException: (LoadError) no such file to load -- bundler/dep_proxy
  while locating org.embulk.jruby.JRubyScriptingModule$ScriptingContainerProvider
  at org.embulk.jruby.JRubyScriptingModule.configure(JRubyScriptingModule.java:47)
  while locating org.jruby.embed.ScriptingContainer

1 error
	at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:466)
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:184)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:110)
	at com.google.inject.Guice.createInjector(Guice.java:96)
	at org.embulk.guice.Bootstrap.start(Bootstrap.java:143)
	at org.embulk.guice.Bootstrap.build(Bootstrap.java:110)
	at org.embulk.guice.Bootstrap.initialize(Bootstrap.java:100)
	at org.embulk.EmbulkEmbed$Bootstrap.build(EmbulkEmbed.java:111)
	at org.embulk.EmbulkEmbed$Bootstrap.initialize(EmbulkEmbed.java:91)
	at org.embulk.EmbulkSetup.setup(EmbulkSetup.java:45)
	at org.embulk.cli.EmbulkRun.runSubcommand(EmbulkRun.java:450)
	at org.embulk.cli.EmbulkRun.run(EmbulkRun.java:100)
	at org.embulk.cli.Main.main(Main.java:28)
Caused by: org.jruby.embed.EvalFailedException: (LoadError) no such file to load -- bundler/dep_proxy
	at org.jruby.embed.internal.EmbedEvalUnitImpl.run(EmbedEvalUnitImpl.java:131)
	at org.jruby.embed.ScriptingContainer.runUnit(ScriptingContainer.java:1307)
	at org.jruby.embed.ScriptingContainer.runScriptlet(ScriptingContainer.java:1300)
	at org.embulk.jruby.JRubyScriptingModule$ScriptingContainerProvider.setBundlerPluginSourceDirectory(JRubyScriptingModule.java:273)
	at org.embulk.jruby.JRubyScriptingModule$ScriptingContainerProvider.get(JRubyScriptingModule.java:164)
	at org.embulk.jruby.JRubyScriptingModule$ScriptingContainerProvider.get(JRubyScriptingModule.java:53)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
	at com.google.inject.internal.BoundProviderFactory.provision(BoundProviderFactory.java:72)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61)
	at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:62)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:145)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:205)
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:199)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:199)
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:180)
	... 11 more
Caused by: org.jruby.exceptions.RaiseException: (LoadError) no such file to load -- bundler/dep_proxy
	at RUBY.block in expand_dependencies(uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:619)
	at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1734)
	at RUBY.block in expand_dependencies(uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:618)
	at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1734)
	at RUBY.expand_dependencies(uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:615)
	at RUBY.converge_locked_specs(uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:583)
	at RUBY.resolve(uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:195)
	at RUBY.specs(uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:140)
	at RUBY.specs_for(uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:185)
	at RUBY.requested_specs(uri:classloader:/gems/bundler-1.10.6/lib/bundler/definition.rb:174)
	at RUBY.requested_specs(uri:classloader:/gems/bundler-1.10.6/lib/bundler/environment.rb:18)
	at RUBY.setup(uri:classloader:/gems/bundler-1.10.6/lib/bundler/runtime.rb:13)
	at RUBY.setup(uri:classloader:/gems/bundler-1.10.6/lib/bundler.rb:127)
	at RUBY.<main>(uri:classloader:/gems/bundler-1.10.6/lib/bundler/setup.rb:8)
	at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:955)
	at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.core_ext.kernel_require.require(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:59)
	at RUBY.<main>(<script>:1)
```

--- a/embulk-core/src/main/java/org/embulk/jruby/JRubyScriptingModule.java
+++ b/embulk-core/src/main/java/org/embulk/jruby/JRubyScriptingModule.java
@@ -269,6 +269,28 @@ public class JRubyScriptingModule
                 jruby.runScriptlet("Gem.clear_paths");
                 jruby.runScriptlet("require 'bundler'");
 
+                // TODO: Remove the monkey patch once the issue is fixed on Bundler or JRuby side.
+                // @see <a href="https://github.com/bundler/bundler/issues/4565">Bundler::SharedHelpers.clean_load_path does cleanup the default load_path on jruby - Issue #4565 - bundler/bundler</a>
+                final String monkeyPatchOnSharedHelpersCleanLoadPath =
+                    "begin\n" +
+                    "  require 'bundler/shared_helpers'\n" +
+                    "  module Bundler\n" +
+                    "    module DisableCleanLoadPath\n" +
+                    "      def clean_load_path\n" +
+                    "        # Do nothing.\n" +
+                    "      end\n" +
+                    "    end\n" +
+                    "    module SharedHelpers\n" +
+                    "      def included(bundler)\n" +
+                    "        bundler.send :include, DisableCleanLoadPath\n" +
+                    "      end\n" +
+                    "    end\n" +
+                    "  end\n" +
+                    "rescue LoadError\n" +
+                    "  # Ignore LoadError.\n" +
+                    "end\n";
+                jruby.runScriptlet(monkeyPatchOnSharedHelpersCleanLoadPath);
+
                 jruby.runScriptlet("Bundler.load.setup_environment");
                 jruby.runScriptlet("require 'bundler/setup'");
                 // since here, `require` may load files of different (newer) embulk versions

